components/emoji_picker/emoji_picker_overlay.tsx

```diff
@@ -1,17 +1,34 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {Overlay} from 'react-bootstrap';
 import memoize from 'memoize-one';
 
-import {popOverOverlayPosition} from 'utils/position_utils.tsx';
+import {popOverOverlayPosition} from 'utils/position_utils';
 import {Constants} from 'utils/constants';
 
+import {Emoji} from '@mattermost/types/emojis';
+
 import EmojiPickerTabs from './emoji_picker_tabs.jsx';
 
-export default class EmojiPickerOverlay extends React.PureComponent {
+type Props = {
+    show: boolean;
+    container?: () => ReactNode | ReactNode;
```

Comment:
```suggestion
    container?: () => ReactNode;
```

Reply:
Hi, @M-ZubairAhmed I referred to React Bootstrap docs before defining types for target and container props. According to React Bootstrap docs, the target and container props can receive DOM nodes and function which returns DOM nodes. What are your thoughts about it?

https://react-bootstrap.github.io/components/overlays/#overlay-props:~:text=API-,%23,-Overlay

Reply:
i see it goes all the way back to here actually https://github.com/react-restart/ui/blob/de7ce33ea3af51d77e589bf51f41ca071b3c0a6f/src/useWaitForDOMRef.ts 
But the correct type seems to be fine.